aboutcitus/concepts.rst

```diff
@@ -0,0 +1,162 @@
+
+The :ref:`pg_dist_shard <pg_dist_shard>` metadata table on the coordinator contains a row for each shard of each distributed table in the system. The row matches a shardid with a range of integers in a hash space (shardminvalue, shardmaxvalue):
+
+.. code-block:: sql
+
+    SELECT * from pg_dist_shard;
+     logicalrelid  | shardid | shardstorage | shardminvalue | shardmaxvalue 
+    ---------------+---------+--------------+---------------+---------------
+     github_events |  102026 | t            | 268435456     | 402653183
+     github_events |  102027 | t            | 402653184     | 536870911
+     github_events |  102028 | t            | 536870912     | 671088639
+     github_events |  102029 | t            | 671088640     | 805306367
+     (4 rows)
+
+If the coordinator node wants to determine which shard holds a row of ``github_events``, it hashes the value of the distribution column in the row, and checks which shard's range contains the hashed value. (The ranges are defined so that the image of the hash function is their disjoint union.)
+
+Suppose that shard 102027 is associated with the row in question. This means the row should be read or written to a table called ``github_events_102027`` in one of the workers. Which worker? That is determined entirely by the metadata tables, and the mapping of shard to worker is known as the shard *placement*.
+
+.. code-block:: sql
+
+  SELECT
+      shardid,
+      node.nodename,
+      node.nodeport
+  FROM pg_dist_placement placement
+  JOIN pg_dist_node node
+    ON placement.groupid = node.groupid
+   AND node.noderole = 'primary'::noderole
+  WHERE shardid = 102027;
+
+  ┌─────────┬───────────┬──────────┐
+  │ shardid │ nodename  │ nodeport │
+  ├─────────┼───────────┼──────────┤
+  │  102027 │ localhost │     5433 │
+  └─────────┴───────────┴──────────┘
+
+Joining some :ref:`metadata tables <metadata_tables>` gives us the answer. These are the types of lookups that the coordinator does to route queries. It rewrites queries into fragments that refer to the specific tables like ``github_events_102027``, and runs those fragments on the appropriate workers.
+
+In our example of ``github_events`` there were four shards. The number of shards is configurable per table at the time of its distribution across the cluster. The best choice of shard count depends on your use case, see :ref:`prod_shard_count`.
+
+Finally note that Citus allows shards to be replicated for protection against data loss. There are two replication "modes:" Citus replication and streaming replication. The former creates extra backup shard placements and runs queries against all of them that update any of them. The latter is more efficient and utilizes PostgreSQL's streaming replication to back up the entire database of each node to a follower database. This is transparent and does not require the involvement of Citus metadata tables.
+
+Co-Location
```

Comment:
The original content you had was sufficient. I think the player example is good, but maybe we can just use the store example? Also, we may not have to show the schema etc, but just verbally call out that if we partition everything by store-id, then all data for a store will be on the same node.

Reply:
(done)